Modules/Core/Common/include/itkPointSet.hxx

```diff
@@ -73,12 +73,79 @@ PointSet<TPixelType, VDimension, TMeshTraits>::SetPoints(PointsVectorContainer *
   {
     itkExceptionMacro("Number of entries in given 1d array incompatible with the point dimension");
   }
+
+  // Note: this cast is unsafe. It may lead to undefined behavior.
   auto * pointsPtr = reinterpret_cast<PointsContainer *>(points);
 
   m_PointsContainer = pointsPtr;
   this->Modified();
 }
 
+
+template <typename TPixelType, unsigned int VDimension, typename TMeshTraits>
+void
+PointSet<TPixelType, VDimension, TMeshTraits>::SetPointsByCoordinates(const std::vector<CoordRepType> & coordinates)
+{
+  itkDebugMacro("Setting the points to the specified coordinates");
+
+  const size_t numberOfCoordinates = coordinates.size();
+
+  if (numberOfCoordinates % PointDimension != 0)
+  {
+    itkExceptionMacro("Number of specified coordinates incompatible with the point dimension");
+  }
+
+  const size_t numberOfPoints = numberOfCoordinates / PointDimension;
+
+  if ((m_PointsContainer != nullptr) && (m_PointsContainer->size() == numberOfPoints))
+  {
+    const auto & points = m_PointsContainer->CastToSTLConstContainer();
+
+    const bool equalCoordinates = [&points, &coordinates] {
+      auto coordinateIterator = coordinates.cbegin();
+
+      for (const auto & point : points)
+      {
+        if (!std::equal(point.cbegin(), point.cend(), coordinateIterator))
```

Comment:
Interesting errors at https://open.cdash.org/viewBuildError.php?buildid=9909255, please have a look:

```
/.../s/Modules/Core/Common/include/itkPointSet.hxx: In instantiation of 'void itk::PointSet<TPixelType, VDimension, TMeshTraits>::SetPointsByCoordinates(const std::vector<typename TMeshTraits::CoordRepType>&) [with TPixelType = double; unsigned int VDimension = 2; TMeshTraits = itk::QuadEdgeMeshTraits<double, 2, bool, bool, float, float>; typename TMeshTraits::CoordRepType = float]':
Wrapping/Modules/ITKQuadEdgeMesh/itkQuadEdgeMeshBasePython.cpp:11546:97:   required from here

Modules/Core/Common/include/itkPointSet.hxx:109:31: error: 'const struct std::pair<const long unsigned int, itk::QuadEdgeMeshPoint<float, 2, itk::GeometricalQuadEdge<long unsigned int, long unsigned int, bool, bool, true> > >' has no member named 'cbegin'
```

It appears that `PointsContainer` may be an `itk::MapContainer`, rather than just an `itk::VectorContainer`! 

https://github.com/InsightSoftwareConsortium/ITK/blob/9a5dade7f7e13fcef307f8224a01c2ce2b0ab13b/Modules/Core/QuadEdgeMesh/include/itkQuadEdgeMeshTraits.h#L87 

Obviously `reinterpret_cast<PointsContainer *>(points)` won't work when `PointsContainer` is an instance of `itk::MapContainer`. But my pull request should also still be adjusted to support `itk::MapContainer`, apparently 🤷